tox.ini

```diff
@@ -32,25 +34,36 @@ commands =
 deps =
     pytest
     pyyaml
+    ruamel.yaml
+    ansible
 commands =
-    bash -c 'if find tests/unit -name \*.py | grep -q . ; then pytest {posargs} tests/unit ; else echo no python files - skipping; fi'
+    bash -c 'if find tests/unit -name \*.py | grep -q . ; then mkdir -p /var/tmp/src/systemrole; mkdir -p /var/tmp/collections; COLLECTION_SRC_PATH=/var/tmp/src COLLECTION_DEST_PATH=/var/tmp/collections COLLECTION_ROLE=systemrole pytest {posargs} tests/unit; rm -rf /var/tmp/src /var/tmp/collections ; else echo no python files - skipping; fi'
 
 [pytest]
 addopts = -rxs -v
 
 [flake8]
 show_source = true
-max-line-length = 88
-ignore = E402,W503
+max-line-length = 120
+ignore = W503
 exclude = .venv,.tox
 statistics = true
 #verbose = 3
 
 [pylint]
-max-line-length = 88
+max-line-length = 120
 
 [pycodestyle]
-max-line-length = 88
+max-line-length = 120
+
+[testenv:collection]
+ansible_python_interpreter = /usr/bin/python3
+deps =
+    ruamel.yaml
+    ansible
+    jmespath
+commands =
+    bash -e -c 'if [ -x lsr_roles2collections.sh ]; then COLLECTION_SRC_PATH=/var/tmp/src COLLECTION_DEST_PATH=/var/tmp/collections ROLES="certificate kdump kernel_settings logging metrics nbde_client tuned" {toxinidir}/lsr_roles2collections.sh; cd /var/tmp/collections/ansible_collections/fedora/system_roles/tests; ANSIBLE_COLLECTIONS_PATHS=/var/tmp/collections:/usr/share/ansible/collections ansible-playbook --become --become-user root -e 'ansible_python_interpreter=/usr/bin/python3' -vvvv tests_default.yml; rm -rf /var/tmp/src /var/tmp/collections; fi'
```

Comment:
I don't think we should be doing this here - tox/travis tests for the auto-maintenance repo should not be pulling in roles repos and running ansible-playbook.  That should be done in each individual role.

If we really want to have a test in this repo that does an actual role2collection conversion of an actual role, then perhaps some sort of dummy role in the tests directory - but I think you already did that in the unit tests.

Reply:
In the current travis env, `roles: [certificate kdump kernel_settings logging metrics nbde_client tuned] ` passed the test. Is it ok to keep it? Or do you think there's no value added to the unit test?

Regarding the integration test in each individual role, I agree we should add there to make sure the changes made on the role do not break the collection format or to test the conversion tool if it handles the roles correctly or not in the roles pr. I'm wondering if it's done in travis, do we have the same non-redhat os issue there, too? Or do we have a good workaround?

Reply:
> Is it ok to keep it? Or do you think there's no value added to the unit test?

I would prefer not to keep it - even though it is working now, it will be a burden to maintain in the future.

> Regarding the integration test in each individual role

There are two types of tests that will be run against each individual role:
* tox/travis test e.g. `cd linux-system-roles/logging ; tox -e collection` - this test will be developed in the `template/` repo and synced to the other roles (or we will use the new https://github.com/linux-system-roles/meta_test/ that @i386x is working on)
* integration tests - we'll need some way to build ROLENAME into a collection then run ROLENAME/tests/tests*.yml against the collection.